### PR TITLE
docs: game def draft

### PR DESCRIPTION
**Pull request by Slesarew**

### CODE REVIEW

File: docs/game-provider.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Game provider
+
+Game providers are specified through 2 layers
```
Comment (Zhenya-Barannik) [inline docs/game-provider.md:11]: You gave some properties of a game provider, but the definition of game provider itself is missing. Also those layers are not well-defined it seems.

File: docs/game-provider.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+### Hardcoded logic
+
+Hardcoded game providers MUST be stored in separate module, all well-documented and independently versioned.
+
+### Game provider spec
+
+Game providers are generated statically at the beginning of simulation and do not hold internal state. Spec file should be a TOML in dedicated subdiractory of the model.
```
Comment (Zhenya-Barannik) [inline docs/game-provider.md:19]: Do you aim for two separate code compilation/execution steps? First one to read configs and build required binary, second one to run the game itself.

File: docs/game-provider.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+## Parameters to specify
+
+- Game types (complex logic with whole game spec stack involved)
+- For each game, frequency parameters (including noise generation) - potentially could have complex logic
+- Connected player pools
```
Comment (Zhenya-Barannik) [inline docs/game-provider.md:25]: But what are player pools?

File: docs/game.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+Templite file should be something human-readable, probably TOML. It determines static parameters of the game used in model and should be stored under special game subdirectory of model.
+
+### Game instance
+
+This is generated by **game provider** during runtime depending dynamically.
+
+## Game specification
+
+Game should specify roles of players and their options and full matrix of outcomes (resource change function on every outcome combination). Game should also specify potentially incomplete knowledge interface for players decision making engine.
```
Comment (Zhenya-Barannik) [inline docs/game.md:23]: Or maybe something like Hamiltonian matrix where each value encodes probability of transition between certain states, for each player? Remember that we have probabilistic model.

**Comment by Zhenya-Barannik** [comment]
Everything else seem clear in general (given you will write more about providers and layers).

File: docs/game-provider.md (hunk 2 above)
Comment (Slesarew) [reply docs/game-provider.md:19]: Yes, these should be separated. We'll have clearly defined holography here then, with configs as boundary conditions and run as internal space.

File: docs/game-provider.md (hunk 1 above)
Comment (Slesarew) [reply docs/game-provider.md:11]: Right, forgot it's in different doc - should be moved here probably?

```suggestion
A game provider is the entity that adds games to runtime's game pool.

This is an abstraction to isolate origins of games.

A number of games is proposed for each tick; they may fail to be executed.

The games should be considered to be played "simultaneously" within a tick.

Game providers are specified through 2 layers
```

File: docs/game-provider.md (hunk 3 above)
Comment (Slesarew) [reply docs/game-provider.md:25]: Should be defined in separate document, but general def is in #3

File: docs/game-provider.md (hunk 1 above)
Comment (Zhenya-Barannik) [reply docs/game-provider.md:11]: So games are close to things that what we call "behaviours" at the moment. If I understand correctly their properties are such: 
1. Games are played in parallel within a tick, not sequentially. 
2. Order in which games are added by provider( or providers) inside a tick's pool is only an implementation detail.
A question appears: how we can choose which of conflicting games should be played? Previously we decided this chronologically.

Comment (Zhenya-Barannik) [reply docs/game-provider.md:11]: Games are not behaviours, they are events that in which players may act with behaviours.

**Review by Zhenya-Barannik — COMMENTED** [review]
All major things are solved.

**Review by Zhenya-Barannik — APPROVED** [review]

File: docs/game.md (hunk 4 above)
Comment (Slesarew) [reply docs/game.md:23]: indeed, but let's try to keep it open for now, maybe we'll want something else

**Pull request merged**